pkg/controller/sesv2/emailidentity/setup.go

```diff
@@ -0,0 +1,261 @@
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclient.BoolValue(resp.VerifiedForSendingStatus) {
+	case true:
+		cr.Status.SetConditions(xpv1.Available())
+	case false:
+		cr.Status.SetConditions(xpv1.Unavailable())
+	default:
+		cr.Status.SetConditions(xpv1.Creating())
+	}
+
+	return obs, nil
+}
+
+func (e *hooks) preCreate(ctx context.Context, cr *svcapitypes.EmailIdentity, obj *svcsdk.CreateEmailIdentityInput) error {
+	if cr.Spec.ForProvider.DkimSigningAttributes != nil && cr.Spec.ForProvider.DkimSigningAttributes.DomainSigningSelector != nil {
+		// Retrieve from Secret Object if DomainSigningPrivateKeySecretRef is available
+		if cr.Spec.ForProvider.DomainSigningPrivateKeySecretRef != nil {
+			pk, err := e.getPrivateKeyFromRef(ctx, cr.Spec.ForProvider.DomainSigningPrivateKeySecretRef)
+			if err != nil {
+				return errors.Wrap(err, "private key retrival failed")
+			}
+			obj.DkimSigningAttributes.DomainSigningPrivateKey = awsclient.String(base64.StdEncoding.EncodeToString([]byte(pk)))
+		}
+	}
+	return nil
+}
+
+func (e *hooks) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	// NOTE: Update operation NOT generated by ACK code-generator
+	cr, ok := mg.(*svcapitypes.EmailIdentity)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// update DKIMAttributes (PutEmailIdentityDKIMSigningAttributes) - Enabled as part of SES for Domain Validation using DKIM Records
+	// NOTE: GetEmailIdentityOutput.DkimSigningAttributes is not supported in aws-sdk-go v1.44.174 - update has to be triggered together with changes in other field e.g. MailFromAttributes
+	// This is done in postCreate stage instead
+
+	// update ConfigurationSetAttributes (PutEmailIdentityConfigurationSetAttributes)
+	configurationSetAttributesInput := &svcsdk.PutEmailIdentityConfigurationSetAttributesInput{
+		ConfigurationSetName: cr.Spec.ForProvider.ConfigurationSetName,
+		EmailIdentity:        cr.Spec.ForProvider.EmailIdentity,
+	}
+	if _, err := e.client.PutEmailIdentityConfigurationSetAttributesWithContext(ctx, configurationSetAttributesInput); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+	}
+
+	// Update MailFromAttributes (PutEmailIdentityMailFromAttributes)
+	// NOTE: Currently MailFromAttributes is not supported in aws-sdk-go v1.44.174 and is declared in Custom-Type
+	if cr.Spec.ForProvider.MailFromAttributes != nil {
+		mailFromAttributesInput := &svcsdk.PutEmailIdentityMailFromAttributesInput{
+			BehaviorOnMxFailure: cr.Spec.ForProvider.MailFromAttributes.BehaviorOnMxFailure,
+			EmailIdentity:       cr.Spec.ForProvider.EmailIdentity,
+			MailFromDomain:      cr.Spec.ForProvider.MailFromAttributes.MailFromDomain,
+		}
+
+		if _, err := e.client.PutEmailIdentityMailFromAttributesWithContext(ctx, mailFromAttributesInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
```

Comment:
There should be a different error message for each `Wrap` to help identify where the error came from. `"update failed"` is too generic here.

Reply:
resolved